packages/js-sdk/src/sandbox/commands/index.ts

```diff
@@ -292,33 +294,51 @@ export class Commands {
 
   /**
    * Start a new command and wait until it finishes executing.
-   * 
+   *
    * @param cmd command to execute.
    * @param opts options for starting the command.
-   * 
+   *
    * @returns `CommandResult` result of the command execution.
    */
   async run(
     cmd: string,
     opts?: CommandStartOpts & { background?: false }
   ): Promise<CommandResult>
+
   /**
    * Start a new command in the background.
    * You can use {@link CommandHandle.wait} to wait for the command to finish and get its result.
-   * 
+   *
    * @param cmd command to execute.
    * @param opts options for starting the command
-   * 
+   *
    * @returns `CommandHandle` handle to interact with the running command.
    */
   async run(
     cmd: string,
-    opts?: CommandStartOpts & { background: true }
+    opts: CommandStartOpts & { background: true }
   ): Promise<CommandHandle>
+
+  // NOTE - The following overload seems redundant, but it's required to make the type inference work correctly.
+
+  /**
+   * Start a new command.
+   *
+   * @param cmd command to execute.
+   * @param opts options for starting the command.
+   *   - `opts.background: true` - runs in background, returns `CommandHandle`
+   *   - `opts.background: false | undefined` - waits for completion, returns `CommandResult`
+   *
+   * @returns Either a `CommandHandle` or a `CommandResult` (depending on `opts.background`).
+   */
+  async run(
+    cmd: string,
+    opts?: CommandStartOpts & { background?: boolean }
+  ): Promise<CommandHandle | CommandResult>
```

Comment:
I don't understand why the overload is requires, can you make it work without pls.